### ISSUE TITLE
Tutorial - Using the WP REST API

### ISSUE THREAD

**Issue by jonathanbossenger** [WordPress/Learn#1269]
# Topic Description
The WordPress REST API provides an interface for applications to interact with your WordPress site by sending and receiving data as JSON (JavaScript Object Notation) objects. It is the foundation of the WordPress Block Editor.

However, it also provides a more predictable and structured way to interact with your site’s content than using admin-ajax. In this session, you will learn how to use the WP REST API in place of using admin-ajax, in order to interact with existing WordPress data.

# Related Resources
Links to related content on Learn, HelpHub, DevHub, GitHub Gutenberg Issues, DevNotes, etc.
- [ ]

# Guidelines
Review the [team guidelines](https://make.wordpress.org/training/handbook/guidelines/)

# Tutorial Development Checklist

- [ ] Vetted by instructional designers for content idea
- [ ] Provide feedback of the idea
- [ ] Gather links to Support and Developer Docs
- [ ] Review any related material on Learn
- [ ] Define several SEO keywords to use in the article and where they should be prominently used
- [ ] Description and Objectives finalized
- [ ] Tutorial [created](https://make.wordpress.org/training/handbook/tutorials/creating-a-tutorial) and announced to the team for Q/A review
- [ ] Tutorial reviewed and [ready to publish](https://make.wordpress.org/training/handbook/tutorials/publishing-a-tutorial)
- [ ] Tutorial [submitted and published to WPTV](https://make.wordpress.org/training/handbook/tutorials/publishing-a-tutorial/#1-submit-your-video-to-wordpress-tv) 
- [ ] Tutorial published on WPTV
- [ ] Tutorial [captioned](https://make.wordpress.org/training/handbook/tutorials/publishing-a-tutorial/#2-generate-subtitles-and-a-transcript-for-your-video) 
- [ ] Tutorial [created on Learn.WordPress.org](https://make.wordpress.org/training/handbook/tutorials/publishing-a-tutorial/#3-create-edit-your-workshop-post)
- [ ] Tutorial post reviewed for grammar, spelling, etc.
- [ ] Tutorial published on Learn.WordPress.org
- [ ] Tutorial [announced to Marketing Team](https://docs.google.com/spreadsheets/d/1rY9gyA0KljjlinPE2B0S2qfsN8Nz24_N_KzqB5CrYZM/edit#gid=469255212) for promotion

**Comment by jonathanbossenger** [comment]
WIP Script - https://github.com/jonathanbossenger/wp-content/blob/main/tutorials/using-wordpress-rest-api.md

**Comment by jonathanbossenger** [comment]
https://user-images.githubusercontent.com/180629/214626816-42e59f58-60ce-4921-a86a-09bfb5c2affa.mp4

Tutorial ready for review

**Comment by wparasae** [comment]
I'm going to review this as a non-developer; I hope that someone with more developer knowledge can spot-check this for content too, but I'll do my best! You definitely know your stuff, Jonathan. Thank you for sharing your knowledge with us! I really appreciate your examples! I've learned several new places I can learn to tinker within WordPress from this. I also like that you keep your transitions simple; you have really mastered streamlining your content.

**An optional change:** The text is a bit small around the 4:00 minute mark and on. Consider if zooming in may make this tutorial a bit easier to read (I'm thinking specifically for folks who may be on mobile) or simply something to keep in mind for future tutorials--I fix this myself quite a bit on my own tutorials and am working on the habit of trying to make screenshots and tutorial videos larger by default, which is why I noticed. I'm labeling this as an optional change, since it's definitely possible to adjust my own screensize to view this.

**At 6:58/6:59** Consider adding a transition there -- it wasn't immediately clear that the screen had shifted.
**At 10:44** Consider adding a transition here. :) 

I really appreciate that you move back and forth between documentation that I could look up myself as I learn this, and your examples. You do a good job of balancing the two, which encourages me to learn more. Goodness, I want to learn how to develop with WordPress now!

This looks great overall. Very polished!

**Comment by jonathanbossenger** [comment]
Thanks for your detailed review @wparasae. I really appreciate the input of a non-developer, because my goal is always to help non-developers learn these topics as well.

> The text is a bit small around the 4:00 minute mark and on. 

This is one I struggle with a lot. I'll look into what I can do there in terms of available space and seeing the code on the screen simultaneously.

> At 6:58/6:59 Consider adding a transition there -- it wasn't immediately clear that the screen had shifted.
> At 10:44 Consider adding a transition here. :)

Will do!

**Comment by nomad-skateboarding-dev** [comment]
## This is a non-developer/novice review. 🙂

### Impression
- This was awesome! Even though some of the same concepts have been trying to enter my brain for years, I sincerely feel that many things just clicked into place after watching this. It's easy to follow when you have the code tree open in the side panel.

- As mentioned in the first review, this content has me excited to learn to code! Great work.
- I'm looking forward to the next tutorial you mentioned at the end of this one! Thank you for what you're contributing to all of us, Jonathan. You're appreciated. 🙌

### Feedback 1
I'm a mobile only user. I'm unable to really view the code starting from 4 minutes on, as you already noted in your reply to the first review, Jonathan. I am in need of new glasses, so it would be visually impaired currently.

My solution was to download the MP4 file and then play it from my mobile device. I was then able to zoom in on the screen, which I could not via mobile browser.

Also as you noted, I'm not sure what a solution would be, as you need to show the code in order to get across what you're doing. Maybe letting people know that the ability to download the video is there as an option? I honestly don't have any good solution to propose here. 

### Feedback 2
Showing the use of MDN Web Docs added to my trust in the information you were giving me. It also added to my feelings of you the presenter, and WordPress as a whole, as "good people". (Full disclosure, I was previously a member of the Mozilla support team for Firefox mobile 🙂)

**Comment by westnz** [comment]
That is a mouthful, and it is just scratching the surface!  Excellent work, Jonathan.  The coding parts are clear and easy to follow.

**One suggestion: 1:05-1:29**
I would have a separate slide showing bullet points of quoted text as you speak.  It would be easier to follow if your words and text on the screen overlap more here.  

Thank you for all the hard work you put into these tutorials!

**Comment by jonathanbossenger** [comment]
> **One suggestion: 1:05-1:29** I would have a separate slide showing bullet points of quoted text as you speak. It would be easier to follow if your words and text on the screen overlap more here.

@westnz the way I recorded this section was as a result of you pointing out that my explanation slides can get too wordy, and you find you're focusing less on what I'm saying, and more on the words on the slide, and getting confused. So I switched to this method, where the slide contains the "main point" and the spoken section fills in the rest.

**Comment by jonathanbossenger** [comment]
https://user-images.githubusercontent.com/180629/218783145-ea31833b-94d4-4f7d-b602-b9ce3c5824dc.mp4

Version 2 updated

**Comment by westnz** [comment]
> > **One suggestion: 1:05-1:29** I would have a separate slide showing bullet points of quoted text as you speak. It would be easier to follow if your words and text on the screen overlap more here.
> 
> @westnz the way I recorded this section was as a result of you pointing out that my explanation slides can get too wordy, and you find you're focusing less on what I'm saying, and more on the words on the slide, and getting confused. So I switched to this method, where the slide contains the "main point" and the spoken section fills in the rest.

Hey Jonathan!  I thought I was saying the same thing again 😄 but maybe I can explain it better in person when we chat again. It is of course very minor.
PS: Great tutorial!!

**Comment by jonathanbossenger** [comment]
> Hey Jonathan! I thought I was saying the same thing again 😄 but maybe I can explain it better in person when we chat again.

Yeah, it would be great to have clarity on what you mean here 😄 I'm going to go ahead and publish this as is for now, but let's make a plan to chat sometime next week.

**Comment by jonathanbossenger** [comment]
https://learn.wordpress.org/tutorial/using-the-wordpress-rest-api/
